### PR TITLE
Rewrite main

### PR DESCRIPTION
**Pull request by asauber**
* Strip out the nested Cobra-command based entrypoint
* Use the latest cloudprovider style entrypoint
* Upgrade dependencies
* Upgrade base Kubernetes version to 1.11.3
* Switch from glide to dep!

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -0,0 +1,63 @@
+# Gopkg.toml example
+#
+# Refer to https://golang.github.io/dep/docs/Gopkg.toml.html
```
Comment (sarahelizgray) [inline Gopkg.toml]: Should we preserve this example code? Is that helpful?

File: Gopkg.toml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+# [[override]]
+#   name = "github.com/x/y"
+#   version = "2.4.0"
+#
+# [prune]
+#   non-go = false
+#   go-tests = true
+#   unused-packages = true
+
+required = ["github.com/json-iterator/go"]
```
Comment (sarahelizgray) [inline Gopkg.toml]: GO dependency n00b question: Do we need this required statement if we then have the same repo specified as a contraint at line 42?

File: Gopkg.toml (hunk 1 above)
Comment (asauber) [reply Gopkg.toml]: Not particularly helpful. I'll remove it.

File: Gopkg.toml (hunk 2 above)
Comment (asauber) [reply Gopkg.toml]: Don't blame you on that. There are over 15 go dependency management systems ([https://github.com/golang/go/wiki/PackageManagementTools](https://github.com/golang/go/wiki/PackageManagementTools)) This one, `dep`, requires you to specify a package in the "required" array if you would like to impose a constraint on a package that is not directly (but indirectly) required by your application. In this case, the constraint solver was picking a bad version for this package (couldn't compile a dependency code branch). So I had to manually override.

**Pull request merged**